src/main/java/com/netgrif/application/engine/pdf/generator/service/PdfDataHelper.java

```diff
@@ -98,16 +97,30 @@ public void generatePdfFields() {
         this.dataGroups = dataService.getDataGroups(taskId, resource.getTextLocale()).getData();
 
         dataGroups.forEach(dataGroup -> {
-            refreshGrid(dataGroup);
-            dataGroup.getFields().getContent().forEach(field -> {
-                        generateField(dataGroup, field);
-                    }
-            );
-            this.lastX = Integer.MAX_VALUE;
+            if (dataGroup.getParentTaskRefId() == null) {
+                refreshGrid(dataGroup);
+                generateFromDataGroup(dataGroup);
+                this.lastX = Integer.MAX_VALUE;
+            }
         });
         Collections.sort(pdfFields);
     }
 
+    private void generateFromDataGroup(DataGroup dataGroup) {
+        Collection<LocalisedField> fields = dataGroup.getFields().getContent();
+        if (dataGroup.getLayout() != null && dataGroup.getLayout().getType() != null && dataGroup.getLayout().getType().equals("grid"))
+            fields = fields.stream().sorted(Comparator.<LocalisedField, Integer>comparing(f -> f.getLayout().getY()).thenComparing(f -> f.getLayout().getX())).collect(Collectors.toList());
```

Comment:
i would preffer if the content of the `if` was wrapped in `{}` braces

Reply:
Corrected.